Error TS2415: Class 'ChatOpenAI<CallOptions>' incorrectly extends base class 'BaseChatModel<CallOptions, AIMessageChunk>'

### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

Compiling a project that uses `ChatOpenAI` from @langchain/openai during a Docker build results in an error. To reproduce:

package.json:

```
  "scripts": {
    "build": "tsc"
  }
``` 

Dockerfile:

```
FROM node:18

WORKDIR /usr/src/app

ENV PORT 8080

RUN npm install

COPY . .
RUN npm run build

CMD ["npm","start"]
```

```bash
docker build --no-cache -t container-name .
```

### Error Message and Stack Trace (if applicable)

```
node_modules/@langchain/openai/dist/chat_models.d.ts(68,22): error TS2415: Class 'ChatOpenAI<CallOptions>' incorrectly extends base class 'BaseChatModel<CallOptions, AIMessageChunk>'.
'getLsParams' is protected in type 'ChatOpenAI<CallOptions>' but public in type 'BaseChatModel<CallOptions, AIMessageChunk>'.
```

### Description

I'm trying to build a project inside of a Docker build process and it fails with the error listed above.

I have to enable `"skipLibCheck": true` in my tsconfig.json in order to get around this problem.

Note that running the same command (`npm run build`, which runs `tsc`) directly in macOS or Windows does not result in this error. It seems to only occur during a docker build.

### System Info

Node: 18
langchain@0.2.4
langsmith@0.1.30
@langchain/openai@0.1.1

Running `docker build --no-cache -t container-name .` command in macOS Terminal.

I can confirm that I am facing a related issue when I am trying to do something like the following:
```
this.client = new ChatOpenAI({
    model: 'gpt-4o',
    temperature: 0,
})
.
.
.
const agent = createToolCallingAgent({
    llm: this.client,  // <---- The point where error occurs (even my TS linter is able to point it out)
    tools,
    prompt,
})
```
I get the following error:
```
error TS2322: Type 'ChatOpenAI<ChatOpenAICallOptions>' is not assignable to type 'BaseChatModel<BaseLanguageModelCallOptions, BaseMessageChunk>'.
  Property 'getLsParams' is protected in type 'ChatOpenAI<ChatOpenAICallOptions>' but public in type 'BaseChatModel<BaseLanguageModelCallOptions, BaseMessageChunk>'
```

--- a/libs/langchain-openai/package.json
+++ b/libs/langchain-openai/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@langchain/openai",
-  "version": "0.1.1",
+  "version": "0.1.2",
   "description": "OpenAI integrations for LangChain.js",
   "type": "module",
   "engines": {

